### PR TITLE
@W-13557285: [Android] Upgrade to Gradle 8.x and AGP 8.x Via RN 0.72.x

### PR DESCRIPTION
**Pull request by JohnsonEricAtSalesforce**
🎸 _**Ready For Final Review**_ 🥁

  Be aware this pull request is one of four related to the React Native 0.72.7, Gradle 8.2 and Android Gradle Plugin 8.2.0 upgrades.
👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2492
👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-CordovaPlugin/pull/631
👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/pull/345
👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Templates/pull/387

  This prepares SalesforceMobileSDK-ReactNative for supporting React Native 0.72.7, Gradle 8.2, Android Gradle Plugin 8.2.0 and Android 34 in the other three repositories listed above.  Overall, the changes here are relatively small though the platform update has significant impacts to downstream projects.

  I tested this primarily by using `forcereact` to generate every React Native template I could conceive of.  Everything compiled and ran OK for both Android and iOS.

### CODE REVIEW

**Comment by wmathurin** [comment]
Also could you subject of the PR reflect the fact it has a RN upgrade?

File: ios/SalesforceReact/SFMobileSyncReactBridge.m [hunk 1]
```diff
@@ -50,8 +50,8 @@ @implementation SFMobileSyncReactBridge
 
 RCT_EXPORT_METHOD(getSyncStatus:(NSDictionary *)args callback:(RCTResponseSenderBlock)callback)
 {
-    NSNumber *syncId = (NSNumber *) [args nonNullObjectForKey:kSyncIdArg];
-    NSString *syncName = (NSString *) [args nonNullObjectForKey:kSyncNameArg];
+    NSNumber *syncId = (NSNumber *) [args sfsdk_nonNullObjectForKey:kSyncIdArg];
```
Comment (JohnsonEricAtSalesforce) [inline ios/SalesforceReact/SFMobileSyncReactBridge.m:53]: @wmathurin or @bbirman - I haven't run these tests before on my workstation, but I noticed I had to update some method names for the test suite to compile.  Is this expected and simply some code that hasn't been updated from upstream changes?  Let me know ⏳

Comment (wmathurin) [reply ios/SalesforceReact/SFMobileSyncReactBridge.m:53]: This [PR](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3666) got rid of deprecated methods for 12.0.

Comment (JohnsonEricAtSalesforce) [reply ios/SalesforceReact/SFMobileSyncReactBridge.m:53]: Thanks, @wmathurin - It looks like I was able to catch all the downstream impacts of the pull request you mentioned here and get a passing run in CI ⭐️

File: iosTests/package.json [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
+    "start": "node node_modules/react-native/local-cli/cli.js start"
+  },
+  "sdkDependencies": {
+    "SalesforceMobileSDK-iOS": "https://github.com/forcedotcom/SalesforceMobileSDK-iOS.git#dev"
+  },
+  "dependencies": {
+    "create-react-class": "^15.7.0",
+    "react": "18.2.0",
+    "react-native": "0.72.7",
+    "react-native-force": "git+https://github.com/JohnsonEricAtSalesforce/SalesforceMobileSDK-ReactNative.git#feature/w-13557285-gradle-8-react-native-0.72.6"
```
Comment (wmathurin) [inline iosTests/package.json:15]: Before merging switch back to forcedotcom/SalesforceMobileSDK-ReactNative#dev

**Review by wmathurin — CHANGES_REQUESTED** [review]

**Review by brandonpage — APPROVED** [review]
LGTM, just remember to switch `JohnsonEricAtSalesforce` repo url back to `forcedotcom`.

**Review by wmathurin — APPROVED** [review]

**Pull request merged**